Fix Filterviews

berfarah opened the pull request:
I was unable to test this before, not having a sandbox for it. This is
also a great reason to have tests :)

It's now working \o/

berfarah committed:
Fix Filterviews

I was unable to test this before, not having a sandbox for it. This is
also a great reason to have tests :)

It's now working \o/

kylealwyn commented on src/commands/filterviews.js:
```diff
@@ -16,7 +18,7 @@ function generateFilterViews({ serviceKey, spreadsheetId, sheetname, range, dir,
             console.log(`[${ sheetname }] cleared existing filter views`);
             return Promise.resolve(sheet);
-          }, function() {
+          }, function(err) {
```
now may be a good time to move these to `.catch` blocks

berfarah replied:
Yeah, really should. I guess this is a good time to do it.

berfarah committed:
Added locale and bundleName detection to Adapters

berfarah committed:
Added middleware (un)flatten bundles

In order to support infinitely nested bundles, we are using a flattened
format for CSV:

`key.otherKey.arrayIndex`

For example:

```js
"mailers.confirmation.paragraph.0"
```

turns into

```json
{
  "mailers": {
    "confirmation": {
      "paragraph": []
    }
  }
}
```

and vice versa.

berfarah committed:
Extracted text notifications into separate file

berfarah commented:
Since I was fixing stuff I figured I'd just fix a bunch of shit.

berfarah commented:
I'd suggest making this `2.2.0` because of the addition of nested objects.

Commit #1 is a patch, #2 is a patch, #3 is a minor and #4 is probably not even a patch

kylealwyn commented on src/commands/bundles.js:
```diff
@@ -1,3 +1,4 @@
+const console = require('../console');
```
not sure about overriding global `console`. now i can't run any simple console.logs with my own values for output. how about `logger`?

berfarah replied:
Yeah, I had that gut feeling, but I think I was feeling a little lazy. I can rename it to logger.

berfarah replied:
Updated

kylealwyn reviewed (approved):
 
